File: backend/api/v1/endpoints/recurring_bookings.py
"""
API endpoints for recurring appointments and series management
"""

from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List, Optional, Dict
from datetime import datetime, date, time
from pydantic import BaseModel, Field, validator

from config.database import get_db
from models.appointment_series import AppointmentSeries, RecurrencePattern, SeriesStatus
from models.appointment import Appointment
from models.client import Client
from models.booking import Service
from services.recurring_appointment_service import RecurringAppointmentService
from utils.logging import log_user_action

router = APIRouter()


# Pydantic models for requests and responses
class CreateSeriesRequest(BaseModel):
    """Request model for creating a recurring appointment series"""

    client_first_name: str = Field(..., min_length=1, max_length=100)
    client_last_name: str = Field(..., min_length=1, max_length=100)
    client_email: str = Field(..., regex=r"^[^@]+@[^@]+\.[^@]+$")
    client_phone: str = Field(..., min_length=10, max_length=20)
    barber_id: int
    service_id: int
    location_id: int
    recurrence_pattern: RecurrencePattern
    preferred_time: str = Field(
        ..., regex=r"^([01]?[0-9]|2[0-3]):[0-5][0-9]$"
    )  # HH:MM format
    start_date: date
    end_date: Optional[date] = None
    max_appointments: Optional[int] = Field(None, ge=1, le=100)
    series_discount_percent: Optional[float] = Field(0.0, ge=0.0, le=50.0)
    series_name: Optional[str] = Field(None, max_length=100)
    interval_weeks: Optional[int] = Field(1, ge=1, le=12)  # For custom patterns
    is_flexible_time: bool = False
    advance_booking_days: int = Field(60, ge=1, le=365)
    notes: Optional[str] = None

    @validator("end_date")
    def validate_end_date(cls, v, values):
        if v and "start_date" in values and v <= values["start_date"]:
            raise ValueError("End date must be after start date")
        return v

    @validator("preferred_time")
    def validate_time_format(cls, v):
        try:
            time.fromisoformat(v)
        except ValueError:
            raise ValueError("Time must be in HH:MM format")
        return v


class SeriesResponse(BaseModel):
    """Response model for appointment series"""

    id: int
    series_token: str
    series_name: Optional[str]
    recurrence_pattern: RecurrencePattern
    preferred_time: str
    start_date: date
    end_date: Optional[date]
    max_appointments: Optional[int]
    status: SeriesStatus
    series_discount_percent: float
    total_appointments_created: int
    total_appointments_completed: int
    next_appointment_date: Optional[date]
    client_name: str
    barber_name: str
    service_name: str
    service_price: float
    discounted_price: float
    created_at: datetime

    class Config:
        from_attributes = True


class SeriesSavingsResponse(BaseModel):
    """Response model for series savings calculation"""

    appointments_in_period: float
    regular_price_per_appointment: float
    discounted_price_per_appointment: float
    regular_total: float
    discounted_total: float
    total_savings: float
    savings_per_appointment: float


class SeriesExclusionRequest(BaseModel):
    """Request model for adding series exclusions"""

    exclusion_date: date
    reason: str = Field(..., max_length=100)
    reschedule_to_date: Optional[date] = None


# API Endpoints
@router.post("/series/create", response_model=SeriesResponse)
async def create_appointment_series(
    request: CreateSeriesRequest, db: Session = Depends(get_db)
):
    """Create a new recurring appointment series"""

    recurring_service = RecurringAppointmentService(db)

    try:
        # Find or create client
        client = (
            db.query(Client)
            .filter(
                Client.email == request.client_email,
                Client.barber_id == request.barber_id,
            )
            .first()
        )

        if not client:
            client = Client(
                first_name=request.client_first_name,
                last_name=request.client_last_name,
                email=request.client_email,
                phone=request.client_phone,
                barber_id=request.barber_id,
                customer_type="new",
                first_visit_date=request.start_date,
                total_visits=0,
                total_spent=0.0,
            )
            db.add(client)
            db.flush()

        # Parse preferred time
        preferred_time = time.fromisoformat(request.preferred_time)

        # Create the series
        series = recurring_service.create_appointment_series(
            client_id=client.id,
            barber_id=request.barber_id,
            service_id=request.service_id,
            location_id=request.location_id,
            recurrence_pattern=request.recurrence_pattern,
            preferred_time=preferred_time,
            start_date=request.start_date,
            end_date=request.end_date,
            max_appointments=request.max_appointments,
            series_discount_percent=request.series_discount_percent,
            series_name=request.series_name,
            interval_weeks=request.interval_weeks,
            is_flexible_time=request.is_flexible_time,
            advance_booking_days=request.advance_booking_days,
            notes=request.notes,
        )

        # Generate initial appointments
        appointments = recurring_service.generate_upcoming_appointments(
            series.id, lookahead_days=request.advance_booking_days
        )

        # Log the action
        log_user_action(
            action="recurring_series_created",
            user_id=None,  # Public endpoint
            details={
                "series_id": series.id,
                "client_email": client.email,
                "recurrence_pattern": request.recurrence_pattern.value,
                "initial_appointments": len(appointments),
            },
        )

        return _format_series_response(series)

    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))


@router.get("/series/{series_token}", response_model=SeriesResponse)
async def get_series_by_token(series_token: str, db: Session = Depends(get_db)):
    """Get appointment series by token"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    return _format_series_response(series)


@router.get("/series/{series_token}/appointments")
async def get_series_appointments(
    series_token: str,
    include_past: bool = Query(False, description="Include past appointments"),
    db: Session = Depends(get_db),
):
    """Get all appointments for a series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    query = db.query(Appointment).filter(Appointment.series_id == series.id)

    if not include_past:
        query = query.filter(Appointment.appointment_date >= date.today())

    appointments = query.order_by(Appointment.appointment_date).all()

    return [
        {
            "id": apt.id,
            "appointment_date": apt.appointment_date,
            "appointment_time": apt.appointment_time.time(),
            "status": apt.status,
            "service_name": apt.service_name,
            "service_revenue": apt.service_revenue,
            "duration_minutes": apt.duration_minutes,
            "client_notes": apt.client_notes,
        }
        for apt in appointments
    ]


@router.post("/series/{series_token}/pause")
async def pause_series(
    series_token: str,
    reason: str = Query("Customer requested pause"),
    db: Session = Depends(get_db),
):
    """Pause an appointment series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    if series.status != SeriesStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Series is not active")

    updated_series = recurring_service.pause_series(series.id, reason)
    return {"message": "Series paused successfully", "status": updated_series.status}


@router.post("/series/{series_token}/resume")
async def resume_series(
    series_token: str,
    reason: str = Query("Customer requested resume"),
    db: Session = Depends(get_db),
):
    """Resume a paused appointment series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    if series.status != SeriesStatus.PAUSED:
        raise HTTPException(status_code=400, detail="Series is not paused")

    updated_series = recurring_service.resume_series(series.id, reason)
    return {"message": "Series resumed successfully", "status": updated_series.status}


@router.post("/series/{series_token}/cancel")
async def cancel_series(
    series_token: str,
    reason: str = Query("Customer requested cancellation"),
    db: Session = Depends(get_db),
):
    """Cancel an appointment series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    if series.status not in [SeriesStatus.ACTIVE, SeriesStatus.PAUSED]:
        raise HTTPException(status_code=400, detail="Series cannot be cancelled")

    updated_series = recurring_service.cancel_series(series.id, reason)
    return {"message": "Series cancelled successfully", "status": updated_series.status}


@router.post("/series/{series_token}/exclusions")
async def add_series_exclusion(
    series_token: str, request: SeriesExclusionRequest, db: Session = Depends(get_db)
):
    """Add an exclusion date to a series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    if series.status != SeriesStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Series is not active")

    exclusion = recurring_service.add_series_exclusion(
        series.id, request.exclusion_date, request.reason, request.reschedule_to_date
    )

    return {
        "message": "Exclusion added successfully",
        "exclusion_date": exclusion.exclusion_date,
        "reason": exclusion.exclusion_reason,
    }


@router.get("/series/calculate-savings", response_model=SeriesSavingsResponse)
async def calculate_series_savings(
    service_id: int,
    recurrence_pattern: RecurrencePattern,
    discount_percent: float = Query(..., ge=0.0, le=50.0),
    duration_months: int = Query(12, ge=1, le=60),
    db: Session = Depends(get_db),
):
    """Calculate potential savings from a recurring series"""

    recurring_service = RecurringAppointmentService(db)

    try:
        savings = recurring_service.calculate_series_savings(
            service_id=service_id,
            recurrence_pattern=recurrence_pattern,
            discount_percent=discount_percent,
            duration_months=duration_months,
        )

        return SeriesSavingsResponse(**savings)

    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))


@router.post("/series/{series_token}/generate-appointments")
async def generate_upcoming_appointments(
    series_token: str,
    lookahead_days: int = Query(60, ge=1, le=365),
    db: Session = Depends(get_db),
):
    """Generate upcoming appointments for a series"""

    recurring_service = RecurringAppointmentService(db)
    series = recurring_service.get_series_by_token(series_token)

    if not series:
        raise HTTPException(status_code=404, detail="Series not found")

    if series.status != SeriesStatus.ACTIVE:
        raise HTTPException(status_code=400, detail="Series is not active")

    appointments = recurring_service.generate_upcoming_appointments(
        series.id, lookahead_days
    )

    return {
        "message": f"Generated {len(appointments)} appointments",
        "appointments_created": len(appointments),
        "next_appointment_date": (
            min(apt.appointment_date for apt in appointments) if appointments else None
        ),
    }


def _format_series_response(series: AppointmentSeries) -> SeriesResponse:
    """Format series for response"""

    client_name = f"{series.client.first_name} {series.client.last_name}"
    barber_name = f"{series.barber.first_name} {series.barber.last_name}"

    return SeriesResponse(
        id=series.id,
        series_token=series.series_token,
        series_name=series.series_name,
        recurrence_pattern=series.recurrence_pattern,
        preferred_time=series.preferred_time.strftime("%H:%M"),
        start_date=series.start_date,
        end_date=series.end_date,
        max_appointments=series.max_appointments,
        status=series.status,
        series_discount_percent=series.series_discount_percent,
        total_appointments_created=series.total_appointments_created,
        total_appointments_completed=series.total_appointments_completed,
        next_appointment_date=series.next_appointment_date,
        client_name=client_name,
        barber_name=barber_name,
        service_name=series.service.name,
        service_price=series.service.base_price,
        discounted_price=series.discounted_price_per_appointment,
        created_at=series.created_at,
    )

File: frontend/src/lib/api/recurring-bookings.ts
/**
 * Recurring Bookings API service
 * Handles recurring appointment series creation and management
 */
import apiClient from './client'
import type { ApiResponse, PaginatedResponse } from './client'

// Enums matching backend
export enum RecurrencePattern {
  DAILY = 'daily',
  WEEKLY = 'weekly',
  BIWEEKLY = 'biweekly',
  MONTHLY = 'monthly',
  CUSTOM = 'custom'
}

export enum SeriesStatus {
  ACTIVE = 'active',
  PAUSED = 'paused',
  CANCELLED = 'cancelled',
  EXPIRED = 'expired'
}

// Core interfaces
export interface RecurringBookingSeries {
  id: number
  series_token: string
  series_name?: string
  recurrence_pattern: RecurrencePattern
  preferred_time: string
  start_date: string
  end_date?: string
  max_appointments?: number
  status: SeriesStatus
  series_discount_percent: number
  total_appointments_created: number
  total_appointments_completed: number
  next_appointment_date?: string
  client_name: string
  barber_name: string
  service_name: string
  service_price: number
  discounted_price: number
  created_at: string
  updated_at?: string
}

export interface CreateSeriesRequest {
  client_first_name: string
  client_last_name: string
  client_email: string
  client_phone: string
  barber_id: number
  service_id: number
  location_id: number
  recurrence_pattern: RecurrencePattern
  preferred_time: string // HH:MM format
  start_date: string // YYYY-MM-DD format
  end_date?: string
  max_appointments?: number
  series_discount_percent?: number
  series_name?: string
  interval_weeks?: number // For custom patterns
  is_flexible_time?: boolean
  advance_booking_days?: number
  notes?: string
}

export interface SeriesAppointment {
  id: number
  appointment_date: string
  appointment_time: string
  status: 'scheduled' | 'confirmed' | 'completed' | 'cancelled' | 'no_show'
  service_name: string
  service_revenue: number
  duration_minutes: number
  client_notes?: string
}

export interface SeriesSavings {
  appointments_in_period: number
  regular_price_per_appointment: number
  discounted_price_per_appointment: number
  regular_total: number
  discounted_total: number
  total_savings: number
  savings_per_appointment: number
}

export interface SeriesExclusion {
  exclusion_date: string
  reason: string
  reschedule_to_date?: string
}

export interface SeriesFilter {
  skip?: number
  limit?: number
  status?: SeriesStatus | SeriesStatus[]
  client_email?: string
  barber_id?: number
  service_id?: number
  recurrence_pattern?: RecurrencePattern
  start_date?: string
  end_date?: string
  search?: string
}

export interface RecurrenceOptionConfig {
  id: RecurrencePattern
  name: string
  description: string
  frequency: string
  defaultDiscount: number
  popular?: boolean
  intervalWeeks: number
  minInterval?: number
  maxInterval?: number
  customizable?: boolean
}

// Pre-defined recurrence options
export const RECURRENCE_OPTIONS: RecurrenceOptionConfig[] = [
  {
    id: RecurrencePattern.WEEKLY,
    name: 'Weekly',
    description: 'Premium maintenance for the sharpest look',
    frequency: 'Every week',
    defaultDiscount: 20,
    popular: true,
    intervalWeeks: 1
  },
  {
    id: RecurrencePattern.BIWEEKLY,
    name: 'Bi-Weekly',
    description: 'Stay fresh with regular grooming',
    frequency: 'Every 2 weeks',
    defaultDiscount: 15,
    popular: true,
    intervalWeeks: 2
  },
  {
    id: RecurrencePattern.MONTHLY,
    name: 'Monthly',
    description: 'Perfect for regular maintenance',
    frequency: 'Every 4 weeks',
    defaultDiscount: 12,
    popular: true,
    intervalWeeks: 4
  },
  {
    id: RecurrencePattern.CUSTOM,
    name: 'Custom Schedule',
    description: 'Set your own interval',
    frequency: 'Custom',
    defaultDiscount: 10,
    intervalWeeks: 4,
    minInterval: 1,
    maxInterval: 12,
    customizable: true
  }
]

export const recurringBookingsService = {
  /**
   * Create a new recurring appointment series
   */
  async createSeries(data: CreateSeriesRequest): Promise<ApiResponse<RecurringBookingSeries>> {
    const response = await apiClient.post<RecurringBookingSeries>('/recurring-bookings/series/create', data)
    return { data: response.data }
  },

  /**
   * Get series by token (public access)
   */
  async getSeriesByToken(token: string): Promise<ApiResponse<RecurringBookingSeries>> {
    const response = await apiClient.get<RecurringBookingSeries>(`/recurring-bookings/series/${token}`)
    return { data: response.data }
  },

  /**
   * Get all series with filtering (admin access)
   */
  async getAllSeries(filters?: SeriesFilter): Promise<PaginatedResponse<RecurringBookingSeries>> {
    const params = new URLSearchParams()
    if (filters?.skip) params.append('skip', filters.skip.toString())
    if (filters?.limit) params.append('limit', filters.limit.toString())
    if (filters?.client_email) params.append('client_email', filters.client_email)
    if (filters?.barber_id) params.append('barber_id', filters.barber_id.toString())
    if (filters?.service_id) params.append('service_id', filters.service_id.toString())
    if (filters?.recurrence_pattern) params.append('recurrence_pattern', filters.recurrence_pattern)
    if (filters?.start_date) params.append('start_date', filters.start_date)
    if (filters?.end_date) params.append('end_date', filters.end_date)
    if (filters?.search) params.append('search', filters.search)

    if (Array.isArray(filters?.status)) {
      filters.status.forEach(status => params.append('status', status))
    } else if (filters?.status) {
      params.append('status', filters.status)
    }

    const response = await apiClient.get<PaginatedResponse<RecurringBookingSeries>>(`/recurring-bookings/series?${params.toString()}`)
    return response.data
  },

  /**
   * Get appointments for a specific series
   */
  async getSeriesAppointments(
    token: string,
    includePast: boolean = false
  ): Promise<ApiResponse<SeriesAppointment[]>> {
    const params = new URLSearchParams()
    if (includePast) params.append('include_past', 'true')

    const response = await apiClient.get<SeriesAppointment[]>(
      `/recurring-bookings/series/${token}/appointments?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Pause a series
   */
  async pauseSeries(token: string, reason?: string): Promise<ApiResponse<{ message: string; status: SeriesStatus }>> {
    const params = new URLSearchParams()
    if (reason) params.append('reason', reason)

    const response = await apiClient.post<{ message: string; status: SeriesStatus }>(
      `/recurring-bookings/series/${token}/pause?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Resume a paused series
   */
  async resumeSeries(token: string, reason?: string): Promise<ApiResponse<{ message: string; status: SeriesStatus }>> {
    const params = new URLSearchParams()
    if (reason) params.append('reason', reason)

    const response = await apiClient.post<{ message: string; status: SeriesStatus }>(
      `/recurring-bookings/series/${token}/resume?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Cancel a series
   */
  async cancelSeries(token: string, reason?: string): Promise<ApiResponse<{ message: string; status: SeriesStatus }>> {
    const params = new URLSearchParams()
    if (reason) params.append('reason', reason)

    const response = await apiClient.post<{ message: string; status: SeriesStatus }>(
      `/recurring-bookings/series/${token}/cancel?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Add exclusion date to series
   */
  async addSeriesExclusion(
    token: string,
    exclusionData: SeriesExclusion
  ): Promise<ApiResponse<{ message: string; exclusion_date: string; reason: string }>> {
    const response = await apiClient.post<{ message: string; exclusion_date: string; reason: string }>(
      `/recurring-bookings/series/${token}/exclusions`,
      exclusionData
    )
    return { data: response.data }
  },

  /**
   * Calculate savings for a potential series
   */
  async calculateSavings(
    serviceId: number,
    recurrencePattern: RecurrencePattern,
    discountPercent: number,
    durationMonths: number = 12
  ): Promise<ApiResponse<SeriesSavings>> {
    const params = new URLSearchParams({
      service_id: serviceId.toString(),
      recurrence_pattern: recurrencePattern,
      discount_percent: discountPercent.toString(),
      duration_months: durationMonths.toString()
    })

    const response = await apiClient.get<SeriesSavings>(
      `/recurring-bookings/series/calculate-savings?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Generate upcoming appointments for a series
   */
  async generateUpcomingAppointments(
    token: string,
    lookaheadDays: number = 60
  ): Promise<ApiResponse<{ message: string; appointments_created: number; next_appointment_date?: string }>> {
    const params = new URLSearchParams({
      lookahead_days: lookaheadDays.toString()
    })

    const response = await apiClient.post<{ message: string; appointments_created: number; next_appointment_date?: string }>(
      `/recurring-bookings/series/${token}/generate-appointments?${params.toString()}`
    )
    return { data: response.data }
  },

  /**
   * Utility functions
   */
  getRecurrenceOption(pattern: RecurrencePattern): RecurrenceOptionConfig | undefined {
    return RECURRENCE_OPTIONS.find(option => option.id === pattern)
  },

  getRecurrenceOptions(): RecurrenceOptionConfig[] {
    return RECURRENCE_OPTIONS
  },

  calculateEstimatedSavings(
    basePrice: number,
    pattern: RecurrencePattern,
    discountPercent: number,
    months: number = 12
  ): SeriesSavings {
    const option = this.getRecurrenceOption(pattern)
    const appointmentsPerYear = 52 / (option?.intervalWeeks || 4)
    const appointmentsInPeriod = (appointmentsPerYear * months) / 12

    const regularPricePerAppointment = basePrice
    const discountedPricePerAppointment = basePrice * (1 - discountPercent / 100)
    const regularTotal = regularPricePerAppointment * appointmentsInPeriod
    const discountedTotal = discountedPricePerAppointment * appointmentsInPeriod

    return {
      appointments_in_period: appointmentsInPeriod,
      regular_price_per_appointment: regularPricePerAppointment,
      discounted_price_per_appointment: discountedPricePerAppointment,
      regular_total: regularTotal,
      discounted_total: discountedTotal,
      total_savings: regularTotal - discountedTotal,
      savings_per_appointment: regularPricePerAppointment - discountedPricePerAppointment
    }
  },

  formatRecurrenceFrequency(pattern: RecurrencePattern, intervalWeeks?: number): string {
    switch (pattern) {
      case RecurrencePattern.DAILY:
        return 'Every day'
      case RecurrencePattern.WEEKLY:
        return 'Every week'
      case RecurrencePattern.BIWEEKLY:
        return 'Every 2 weeks'
      case RecurrencePattern.MONTHLY:
        return 'Every 4 weeks'
      case RecurrencePattern.CUSTOM:
        return intervalWeeks ? `Every ${intervalWeeks} week${intervalWeeks > 1 ? 's' : ''}` : 'Custom schedule'
      default:
        return 'Unknown pattern'
    }
  },

  getNextAppointmentDate(
    lastDate: string,
    pattern: RecurrencePattern,
    intervalWeeks: number = 4
  ): Date {
    const lastAppointment = new Date(lastDate)
    const option = this.getRecurrenceOption(pattern)
    const weeksToAdd = pattern === RecurrencePattern.CUSTOM ? intervalWeeks : (option?.intervalWeeks || 4)

    return new Date(lastAppointment.getTime() + (weeksToAdd * 7 * 24 * 60 * 60 * 1000))
  },

  validateSeriesData(data: Partial<CreateSeriesRequest>): { valid: boolean; errors: string[] } {
    const errors: string[] = []

    if (!data.client_first_name?.trim()) errors.push('First name is required')
    if (!data.client_last_name?.trim()) errors.push('Last name is required')
    if (!data.client_email?.trim()) errors.push('Email is required')
    else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(data.client_email)) errors.push('Invalid email format')
    if (!data.client_phone?.trim()) errors.push('Phone number is required')
    if (!data.barber_id) errors.push('Barber selection is required')
    if (!data.service_id) errors.push('Service selection is required')
    if (!data.location_id) errors.push('Location is required')
    if (!data.recurrence_pattern) errors.push('Recurrence pattern is required')
    if (!data.preferred_time?.trim()) errors.push('Preferred time is required')
    else if (!/^([01]?[0-9]|2[0-3]):[0-5][0-9]$/.test(data.preferred_time)) errors.push('Invalid time format')
    if (!data.start_date?.trim()) errors.push('Start date is required')
    else if (new Date(data.start_date) <= new Date()) errors.push('Start date must be in the future')

    if (data.end_date && new Date(data.end_date) <= new Date(data.start_date || '')) {
      errors.push('End date must be after start date')
    }

    if (data.series_discount_percent !== undefined) {
      if (data.series_discount_percent < 0 || data.series_discount_percent > 50) {
        errors.push('Discount must be between 0% and 50%')
      }
    }

    if (data.max_appointments !== undefined) {
      if (data.max_appointments < 1 || data.max_appointments > 100) {
        errors.push('Max appointments must be between 1 and 100')
      }
    }

    return { valid: errors.length === 0, errors }
  }
}
